src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -80,8 +80,38 @@ public function findSpecifiedType(
 					'trait_exists',
 					'enum_exists',
 				], true)) {
+					if (count($node->getArgs()) < 1) {
+						return null;
+					}
+					$argType = $scope->getType($node->getArgs()[0]->value);
+					if (count($argType->getConstantStrings()) !== 1) {
+						return null;
+					}
+					$argConstantString = $argType->getConstantStrings()[0];
+					if (!$this->reflectionProvider->hasClass($argConstantString->getValue())) {
+						return null;
+					}
```

Comment:
you are missing things like

```
$c = _Enum::class;
if (rand(0,1)) {
  $c = _Class::class;
}
class_exists($c);
```

Reply:
Yeah, I dont want to cover all edgecases..